### PR TITLE
[PIL-715][AV] - Corporate Position within group question

### PR DESCRIPTION
**Pull request by aashik-varahan**

### CODE REVIEW

File: app/pages/rfmCorporatePositionPage.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package pages
+
+import models.rfm.CorporatePosition
+import play.api.libs.json.JsPath
+
+case object rfmCorporatePositionPage extends QuestionPage[CorporatePosition] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (jassalrichy) [inline app/pages/rfmCorporatePositionPage.scala:24]: Ideally we should be grouping pages for features in a sub folders. For example:
override def path: JsPath = JsPath \ "rfm" \ toString

File: app/pages/rfmSecurityCheckPage.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object rfmSecurityCheckPage extends QuestionPage[String] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (jassalrichy) [inline app/pages/rfmSecurityCheckPage.scala:23]: Ideally we should be grouping pages for features in a sub folders. For example:
override def path: JsPath = JsPath \ "rfm" \ toString

File: test/controllers/rfm/CorporatePositionControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+
+        redirectLocation(result).value mustEqual controllers.routes.UnderConstructionController.onPageLoad.url
+      }
+    }
+
+    "must redirect to the under construction page when valid data is submitted" in {
```
Comment (jassalrichy) [inline test/controllers/rfm/CorporatePositionControllerSpec.scala:85]: Need a test for when CorporatePosition.NewNfm is selected.

File: test/controllers/rfm/CorporatePositionControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import scala.concurrent.Future
+
+class CorporatePositionControllerSpec extends SpecBase {
+
+  val formProvider = new RfmCorporatePositionFormProvider()
+
+  "RFM Corporate Position controller" when {
+
+    "must return OK and the correct view for a GET" in {
+      val ua = emptyUserAnswers
```
Comment (jassalrichy) [inline test/controllers/rfm/CorporatePositionControllerSpec.scala:42]: Also need a test for when there user answer exists for this page on pageload.

File: test/controllers/rfm/CorporatePositionControllerSpec.scala (hunk 3 above)
Comment (aashik-varahan) [reply test/controllers/rfm/CorporatePositionControllerSpec.scala]: Added a test case when CorporatePosition.NewNfm is selected as valid data.

File: test/controllers/rfm/CorporatePositionControllerSpec.scala (hunk 4 above)
Comment (aashik-varahan) [reply test/controllers/rfm/CorporatePositionControllerSpec.scala:42]: Added a test case when user answer exists for this page on pageload

File: app/controllers/rfm/CorporatePositionController.scala [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  val controllerComponents:  MessagesControllerComponents,
+  view:                      CorporatePositionView,
+  sessionData:               SessionData
+)(implicit ec:               ExecutionContext, appConfig: FrontendAppConfig)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
```
Comment (jassalrichy) [inline app/controllers/rfm/CorporatePositionController.scala:52]: replace  identify with rfmIdentify

File: app/controllers/rfm/CorporatePositionController.scala [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        case Some(value) => form.fill(value)
+        case None        => form
+      }
+      Ok(view(preparedForm, mode))
+    } else {
+      Redirect(controllers.routes.UnderConstructionController.onPageLoad)
+    }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
```
Comment (jassalrichy) [inline app/controllers/rfm/CorporatePositionController.scala:65]: replace identify with rfmIdentify

File: app/controllers/rfm/CorporatePositionController.scala (hunk 5 above)
Comment (jassalrichy) [inline app/controllers/rfm/CorporatePositionController.scala:52]: replace identify with rfmIdentify

File: app/controllers/rfm/SecurityCheckController.scala [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  formProvider:              RfmSecurityCheckFormProvider,
+  val controllerComponents:  MessagesControllerComponents,
+  view:                      SecurityCheckView
+)(implicit ec:               ExecutionContext, appConfig: FrontendAppConfig)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode = NormalMode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
```
Comment (jassalrichy) [inline app/controllers/rfm/SecurityCheckController.scala:49]: replace identify with rfmIdentify

File: app/controllers/rfm/SecurityCheckController.scala [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        case Some(v) => form.fill(v)
+        case None    => form
+      }
+      Ok(view(preparedForm, mode))
+    } else {
+      Redirect(controllers.routes.UnderConstructionController.onPageLoad)
+    }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
```
Comment (jassalrichy) [inline app/controllers/rfm/SecurityCheckController.scala:62]: replace identify with rfmIdentify

**Review by jassalrichy — DISMISSED** [review]

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**